master/buildbot/process/build.py

```diff
@@ -577,37 +578,26 @@ def addStepsAfterLastStep(self, step_factories):
         # Add the new steps to the end.
         self.steps.extend(self.setupBuildSteps(step_factories))
 
-    def getNextStep(self):
+    def getNextStep(self) -> IBuildStep | None:
         """This method is called to obtain the next BuildStep for this build.
-        When it returns None (or raises a StopIteration exception), the build
```

Comment:
I think it makes sense to add to commit description that the behavior of handling StopIteration exception is not documented, thus it's fine to consider this internal API and change it without deprecation.